### PR TITLE
fix #480: type message.id optional for new messages

### LINKED ISSUE #480: Is the `id` on message now required?
https://github.com/GetStream/stream-chat-js/blob/8efa2272be22331f51bcc7b54cb256e52b95c43f/src/types.ts#L1354

Typescript 
image

According with the documentation here https://getstream.io/chat/docs/silent_messages/?language=js we should be able to send a message without and `id`.

image

If it is now required can we use a random uuid for this?

cc @tbarbugli 

gz#6914

### PR DESCRIPTION
**Pull request by mahboubii**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: test/typescript/unit-test.ts [hunk 1]
```diff
@@ -228,6 +228,8 @@ voidReturn = channel.off(eventHandler);
 voidReturn = channel.on('message.new', eventHandler);
 voidReturn = channel.off('message.new', eventHandler);
 
+channel.sendMessage({ text: '' }); // send a msg without id
```
Comment (ferhatelmas) [inline test/typescript/unit-test.ts:231]: could you add some text instead of empty? We might start validating it.

Comment (mahboubii) [reply test/typescript/unit-test.ts:231]: this is just a type check

Comment (ferhatelmas) [reply test/typescript/unit-test.ts:231]: yup, for consistency

Comment (mahboubii) [inline test/typescript/unit-test.ts:231]:
```suggestion
channel.sendMessage({ text: 'text' }); // send a msg without id
```

**Review by ferhatelmas — APPROVED** [review]
LGTM

**Pull request closed without merging**